File: packages/ios-swift/Sources/CinacoinAppKitAuth/EmailLogin.swift
import Foundation

/// Email-based login manager
public final class EmailLoginManager: @unchecked Sendable {
    
    private let authUrl: String
    private let projectId: String
    
    public init(authUrl: String = "https://auth.cinacoin.com", projectId: String) {
        self.authUrl = authUrl
        self.projectId = projectId
    }
    
    /// Register with email and password
    public func register(email: String, password: String) async throws -> AuthResult {
        let url = URL(string: "\(authUrl)/api/auth/register")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.httpBody = try JSONEncoder().encode([
            "email": email,
            "password": password,
            "project_id": projectId
        ])
        
        let (data, response) = try await URLSession.shared.data(for: request)
        
        guard let httpResponse = response as? HTTPURLResponse,
              (200...299).contains(httpResponse.statusCode) else {
            throw AuthError.networkError("Registration failed")
        }
        
        return try JSONDecoder().decode(AuthResult.self, from: data)
    }
    
    /// Login with email and password
    public func login(email: String, password: String) async throws -> AuthResult {
        let url = URL(string: "\(authUrl)/api/auth/login")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.httpBody = try JSONEncoder().encode([
            "email": email,
            "password": password,
            "project_id": projectId
        ])
        
        let (data, response) = try await URLSession.shared.data(for: request)
        
        guard let httpResponse = response as? HTTPURLResponse,
              (200...299).contains(httpResponse.statusCode) else {
            throw AuthError.invalidCredentials
        }
        
        return try JSONDecoder().decode(AuthResult.self, from: data)
    }
    
    /// Send password reset email
    public func sendPasswordReset(email: String) async throws {
        let url = URL(string: "\(authUrl)/api/auth/reset-password")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.httpBody = try JSONEncoder().encode([
            "email": email,
            "project_id": projectId
        ])
        
        let (_, response) = try await URLSession.shared.data(for: request)
        
        guard let httpResponse = response as? HTTPURLResponse,
              (200...299).contains(httpResponse.statusCode) else {
            throw AuthError.networkError("Password reset request failed")
        }
    }
    
    /// Send email verification
    public func sendVerification(email: String) async throws {
        let url = URL(string: "\(authUrl)/api/auth/verify-email")!
        var request = URLRequest(url: url)
        request.httpMethod = "POST"
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.httpBody = try JSONEncoder().encode([
            "email": email,
            "project_id": projectId
        ])
        
        let (_, response) = try await URLSession.shared.data(for: request)
        
        guard let httpResponse = response as? HTTPURLResponse,
              (200...299).contains(httpResponse.statusCode) else {
            throw AuthError.networkError("Verification email send failed")
        }
    }
}

File: packages/flutter-dart/lib/appkit_auth/email_login.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'models/auth_result.dart';
import 'models/auth_error.dart';

/// Email-based login manager
class EmailLoginManager {
  final String authUrl;
  final String projectId;

  EmailLoginManager({
    this.authUrl = 'https://auth.cinacoin.com',
    required this.projectId,
  });

  /// Register with email and password
  Future<AuthResult> register({
    required String email,
    required String password,
  }) async {
    final response = await http.post(
      Uri.parse('$authUrl/api/auth/register'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'email': email,
        'password': password,
        'project_id': projectId,
      }),
    );

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw const NetworkAuthError('Registration failed');
    }

    return AuthResult.fromJson(jsonDecode(response.body));
  }

  /// Login with email and password
  Future<AuthResult> login({
    required String email,
    required String password,
  }) async {
    final response = await http.post(
      Uri.parse('$authUrl/api/auth/login'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'email': email,
        'password': password,
        'project_id': projectId,
      }),
    );

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw const InvalidCredentialsError();
    }

    return AuthResult.fromJson(jsonDecode(response.body));
  }

  /// Send password reset email
  Future<void> sendPasswordReset({required String email}) async {
    final response = await http.post(
      Uri.parse('$authUrl/api/auth/reset-password'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'email': email,
        'project_id': projectId,
      }),
    );

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw const NetworkAuthError('Password reset failed');
    }
  }

  /// Send email verification
  Future<void> sendVerification({required String email}) async {
    final response = await http.post(
      Uri.parse('$authUrl/api/auth/verify-email'),
      headers: {'Content-Type': 'application/json'},
      body: jsonEncode({
        'email': email,
        'project_id': projectId,
      }),
    );

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw const NetworkAuthError('Verification email failed');
    }
  }
}
